bottlerocket/agents/src/bin/eks-resource-agent/eks_provider.rs

```diff
@@ -108,6 +110,180 @@ impl AwsClients {
     }
 }
 
+enum ClusterConfig {
+    Args {
+        cluster_name: String,
+        region: String,
+        version: Option<K8sVersion>,
+        zones: Option<Vec<String>>,
+    },
+    ConfigPath {
+        cluster_name: String,
+        region: String,
+    },
+}
+
+impl ClusterConfig {
+    pub fn new(
+        cluster_name: &Option<String>,
+        region: &Option<String>,
+        version: &Option<K8sVersion>,
+        zones: &Option<Vec<String>>,
+        encoded_config: &Option<String>,
+    ) -> ProviderResult<Self> {
+        let config = match (cluster_name, encoded_config) {
+            (_, Some(eksctl_config)) => {
+                let decoded_config = base64::decode(eksctl_config)
+                    .context(Resources::Clear, "Unable to decode eksctl configuration.")?;
+
+                let config: Value =
+                    serde_yaml::from_str(std::str::from_utf8(&decoded_config).context(
+                        Resources::Clear,
+                        "Unable to convert decoded config to string.",
+                    )?)
+                    .context(Resources::Clear, "Unable to serialize eksctl config.")?;
+
+                let config_path = Path::new(TEST_CLUSTER_CONFIG_PATH);
+                write(config_path, decoded_config).context(
+                    Resources::Clear,
+                    format!(
+                        "Unable to write eksctl configuration to '{}'",
+                        config_path.display()
+                    ),
+                )?;
+
+                let (cluster_name, region) = config
+                    .get("metadata")
+                    .map(|metadata| {
+                        (
+                            metadata.get("name").and_then(|name| name.as_str()),
+                            metadata.get("region").and_then(|region| region.as_str()),
+                        )
+                    })
+                    .context(Resources::Clear, "Metadata is missing from eksctl config.")?;
+
+                Self::ConfigPath {
+                    cluster_name: cluster_name
+                        .context(
+                            Resources::Clear,
+                            "The cluster's name was not in the eksctl config.",
+                        )?
+                        .to_string(),
+                    region: region
+                        .context(
+                            Resources::Clear,
+                            "The cluster's region was not in the eksctl config.",
+                        )?
+                        .to_string(),
+                }
+            }
+            (Some(cluster_name), None) => Self::Args {
+                cluster_name: cluster_name.to_string(),
+                region: region
+                    .as_ref()
+                    .cloned()
```

Comment:
Can this be `to_owned()`?
```suggestion
                    .to_owned()
```